core/bindingnode.cpp

```diff
@@ -118,12 +118,11 @@ bool GammaRay::BindingNode::isPartOfBindingLoop() const
     if (m_foundBindingLoop) {
         return true;
     }
-    for (const auto &dependency : m_dependencies) {
-        if (dependency->isPartOfBindingLoop()) {
-            return true;
-        }
-    }
-    return false;
+
+    bool dependencyIsPartOfBindingLoop = std::any_of(m_dependencies.begin(), m_dependencies.end(),
```

Comment:
just return the result directly? the huge advantage of std algorithms is that their name already makes the intention pretty clear